pkg/gofr/http/responder.go

```diff
@@ -58,6 +64,54 @@ func (r Responder) Respond(data any, err error) {
 	_ = json.NewEncoder(r.w).Encode(resp)
 }
 
+func (r Responder) determineResponse(data any, err error) (statusCode int, errObj any) {
+	// Handle empty struct case first
+	if err != nil && isEmptyStruct(data) {
+		return http.StatusInternalServerError, createErrorResponse(errEmptyResponse)
+	}
+
+	statusCode, errorObj := getStatusCode(r.method, data, err)
+
+	// Ensure valid status code
+	if statusCode == 0 {
+		statusCode = http.StatusInternalServerError
+
+		if errorObj == nil {
+			errorObj = createErrorResponse(errMissingStatusCode)
+		}
+	}
+
+	return statusCode, errorObj
```

Comment:
What about this?

```suggestion
	statusCode, errorObj := getStatusCode(r.method, data, err)

	// Ensure valid status code
	if statusCode == 0 {
		statusCode = http.StatusInternalServerError

		if errorObj == nil {
			errorObj = createErrorResponse(errMissingStatusCode)
		}
	}

	if errorObj == nil {
			errorObj = createErrorResponse(errMissingError)
		}

	return statusCode, errorObj
```

I'm unsure there is a need, but I prefer to raise the point

Reply:
@ccoVeille there is no need of this change because:


- The current implementation follows REST API best practice where successful responses don't include error fields
- The json:"error,omitempty" tag is intentionally designed to omit the error field when nil.